[Tutorial 2.2] 不動産売買アプリのUI模写

k-shimizu3 opened the pull request:
### 課題
 - Tutorial2.2
 - 不動産売買アプリのUI模写

---

### 主な実装内容
 - lib/main.dart
不動産売買アプリへの遷移ボタンを追加

 - lib/residence/residence_screen.dart
不動産売買アプリのUIを作成
個々のUIパーツは 下記 components ディレクトリ内の各ファイルで作成してします。

 - lib/residence/components/
residence_screen.dartで使用するUIパーツを作成してします。

---
### 完成イメージ
<img width="396" alt="screenshot" src="https://user-images.githubusercontent.com/114896398/195234405-c01a6e4d-c3ad-4475-9286-32ecf3f09e1a.png">

---

### 課題（悩んだこと）
これは質問した方がいいかなぁと思ったことを数時間後には忘れてしまっているので、もっとまめにメモを取りたいと思います。

---

### 備考
レビューよろしくお願いします！

k-shimizu3 committed:
AppBarの作成

実装内容
1. ResidenceScreenへの遷移

2. ResidenceScreenのAppBarの作成
・具体的な実装は、lib/residence/components/residence_app_bar.dart
・パッケージ badges の使用

k-shimizu3 committed:
BottomNavigationBarの追加

実装は
lib/redidence/components/residence_bottom_navigation_bar.dart
で行っています。

k-shimizu3 committed:
FloatingActionButtonの追加

実装:
lib/residence/components/residence_floating_action_button.dart

k-shimizu3 committed:
Bodyの実装

k-shimizu3 committed:
Merge branch 'develop' into feature/residence_screen

以下の衝突を解決しました
# Conflicts:
#	lib/main.dart
#	pubspec.yaml

k-shimizu3 committed:
コードの整理

コードの整理を実施しました

k-shimizu3 committed:
DefaultTextStyleを追加して、コード量を減らす

onishitaisei commented:
LGTM!

t-nakata commented on lib/residence/components/category_content.dart:
```diff
@@ -0,0 +1,183 @@
+  final String _favoriteButtonText = 'お気に入り';
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
```
`build`メソッドが150行あります。
長すぎて見通しが悪くなっているので分割してください。

一般的にメソッドの長さの目安としては最大80行、クラスの長さの目安は400行といわれています。
それを超えるようであれば分割を検討してください。

この場合でいうと、画像エリア、テキストエリア、ボタンエリアと3エリアに分けて見てはどうかと思いました。

```
child: Column(
  _buildImages(),
  _buildTexts(),
  _buildButtons(),
)
```

k-shimizu3 replied:
ありがとうございます！
修正してみます！

分ける際に、クラスを作る、クラス内にメソッドを作る、グローバルな関数を作る、の選択肢があると思いますが、何か指針みたいなものはあるのでしょうか？

今回の場合、クラスに分けるほどではないから、クラス内にプライベートメソッドを作るという理解で合ってますでしょうか？

onishitaisei replied:
@kenji-axia 
こちら参考になるのでご確認ください！
今回の場合状態を持つことがないため新規クラスを作るメリットをあまり感じないかもしれないですが、新規クラスを作る方が無駄な描画を防ぐ、見通しを良くするという点でもヘルパーメソッドよりメリットが多いかなと思います。

https://youtu.be/IOyq-eTRhvo

k-shimizu3 replied:
@onishitaisei 
ありがとうございます！
確認してみます！！

k-shimizu3 replied:
@t-nakata 
アドバイスいただいた点を修正しましたので、お手隙の際にご確認お願いします！

修正点：
[CategoryContent() のbuildメソッドの分割](https://github.com/kenji-axia/flutter_tutorial/pull/4/commits/dc148d8ca9dfd8d05d1154d3799c9e3dd93832c7)

k-shimizu3 committed:
CategoryContent() のbuildメソッドの分割

CategoryContent() のbuildメソッドの内容
・画像を横並びに表示
・建物情報の表示
・興味なし / お気に入り ボタンの表示
これらを各Widgetに分割しました。

実装
lib/residence/components/内で
ディレクトリ category_content_parts/ を作成し、分割した各Widgetのファイルを格納しています。
